### PR TITLE
Connect application to an existing API to create and remove books

### PR DESCRIPTION
**Pull request by lenkon**
Project name: Bookstore - connect to API

#### Project description: 
In this project, the React & Redux application will be connected to an existing API to create and remove books from a server.

#### Project requirements:
In the previous project you developed the **add book** and **remove book** features directly in your application. Now you will refactor your code to use the **Bookstore API** to add and remove books.

- Make use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as your backend
- Use either [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [`axios`](https://axios-http.com/docs/intro) with [`createAsyncThunk`](https://redux-toolkit.js.org/api/createAsyncThunk) to make API calls
- Fetch your `books` from Bookstore API on when the component has rendered with `useEffect`
- Refactor your **add book** and **remove book** features to persist your changes in the server
  - Modify your reducer logic to **include HTTP requests to the Bookstore API**
- Styling is not required
- _Everything should work the same as before_

Project requirements link: https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connect_to_api.md

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,53 +1,38 @@
...
 });
 
-export const removeBook = (itemId) => ({
-  type: REMOVE_BOOK, itemId,
-});
+export const fetchBooks = () => (dispatch) => {
+  fetch(URL)
+    .then((response) => response.json())
+    .then((data) => dispatch(getBook(data)));
+};
```
Comment (ghost) [inline src/redux/books/booksSlice.js]: - [ ] as required in the project requirement, Kindly use `createAsyncThunk` to make API calls. Check out this [link](https://redux-toolkit.js.org/api/createAsyncThunk) for more info.

File: src/redux/store.js [hunk 2]
```diff
@@ -1,12 +1,15 @@
 import { configureStore } from '@reduxjs/toolkit';
+// import { applyMiddleware } from 'redux';
+// import thunk from 'redux-thunk';
```
Comment (ghost) [inline src/redux/store.js]: - [ ] Kindly remove commented out code to keep your code base clean

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @lenkon , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ Correct use of Github workflow
✅ No linter error



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by lenkon** [comment]
Hi @frank1738 , 

Thanks for your valuable feedback. I have now used createAsyncThunk to make API calls and also have removed commented out code.

Best regards,
Konjengbam Inoulen Shanongba

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @lenkon  😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done on making the required changes from the previous reviewer
- Excellent communication

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**